src/core/mount-setup.c

```diff
@@ -300,7 +301,8 @@ int mount_cgroup_controllers(char ***join_controllers) {
 
                                         t = set_remove(controllers, *i);
                                         if (!t) {
-                                                free(*i);
+                                                if (has_argument)
+                                                        free(*i);
```

Comment:
`i` seems to be memory allocated from the heap by `malloc`, so I would say that the commit message is a little misleading.

Reply:
Yeah, it should say "stack-allocated".

Reply:
Yeah, sorry. I copied the text from the issue. Please fix the commit message on merge.

Reply:
(and sorry for being a smart-ass about this ;-))